src/network/networkd-link.c

```diff
@@ -2720,6 +2721,7 @@ static int link_new(Manager *manager, sd_netlink_message *message, Link **ret) {
                 .mdns = _RESOLVE_SUPPORT_INVALID,
                 .dnssec_mode = _DNSSEC_MODE_INVALID,
                 .dns_over_tls_mode = _DNS_OVER_TLS_MODE_INVALID,
+                .sysctl_shadow = hashmap_new(&string_hash_ops),
```

Comment:
@teknoraver Not fixed?

Reply:
Why lazily initialization is preferred over ahead of time?
I'm very sure that we write at least an element in the hashmap.

Reply:
we generally do it that way in our tree. This started when unit objects in systemd gained one hash table for each dep type. These are typically quite sparsely populated, hence it became a lot more efficient when initializing the hashmap lazily and systematically treating a NULL hash table like an empty one. We nowadays this across our tree, and it saves quite some allocations in many cases, even though in some it might not.

Hence, the specific case is not in focus here, it's just a matter of coding style: we try to get rid of ahead-of-time allocations of the hashmap constructors througout our codebsae.

Reply:
I mean, to say it differently: it never really makes things worse, it will sometimes make them better though, quite often quite substantially.

Reply:
> Please use `string_hash_ops_free_free` so that both key and value are automatically freed on destruction

In the commit before, I'm using `path_hash_ops_free_free`